sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -330,4 +336,134 @@ public class ChatClient {
+                languageId: RegistrarSettings.languageId,
+                platform: RegistrarSettings.platform,
+                platformUIVersion: RegistrarSettings.platformUIVersion,
+                templateKey: RegistrarSettings.templateKey
+            )
+
+            // Path is device token
+            let transport = RegistrarTransport(
+                ttl: RegistrarSettings.ttl,
+                path: deviceToken,
+                context: RegistrarSettings.context
+            )
+
+            // Register for push notifications
+            registrarClient.setRegistration(with: clientDescription, for: [transport]) { result in
+                switch result {
+                case .success(let response):
+                    self.pushNotificationsStarted = true
+                    completionHandler(.success(response))
+                case .failure(let error):
+                    self.options.logger.error("Failed to start push notifications with error: \(error.localizedDescription)")
+                    completionHandler(.failure(AzureError.client("Failed to start push notifications", error)))
+                }
+            }
+        } catch {
+            options.logger.error("Failed to start push notifications with error: \(error.localizedDescription)")
+            completionHandler(.failure(AzureError.client("Invalid URL.", error)))
+        }
+    }
+
+
+    /// Stop push notifications.
+    /// - Parameter completionHandler: Success indicates push notifications have been stopped.
+    public func stopPushNotifications(
+        completionHandler: @escaping (Result<HTTPResponse?, AzureError>) -> Void
+    ) {
+        guard pushNotificationsStarted else {
+            completionHandler(.failure(AzureError.client("Push notifications are not enabled.")))
+            return
+        }
+
+        guard let registrarClient = registrarClient else {
+            completionHandler(.failure(AzureError.client("Failed to stop push notifications. There is no registrarClient.")))
+            return
+        }
+        
+        registrarClient.deleteRegistration { result in
+            switch result {
+            case .success(let response):
+                self.pushNotificationsStarted = false
+                completionHandler(.success(response))
+            case .failure(let error):
+                self.options.logger.error("Failed to stop push notifications with error: \(error.localizedDescription)")
+                completionHandler(.failure(AzureError.client("Failed to stop push notifications", error)))
+            }
+        }
+    }
+
+    /// Handle the data payload for an incoming push notification.
+    /// - Parameters:
+    ///   - notification: The APNS push notification.
+    ///   - completionHandler: Receives the PushNotificationEvent from the push notification payload.
+    public func handlePush(
+        notification: [AnyHashable : Any],
+        completionHandler: PushNotificationEventHandler
```

Comment:
Can you remind me how this is used by the customer? Is there a sample you can link to?

Reply:
Also, this seems like an eventHandler, not a completionHandler.

Reply:
Kim declared the PushNotificationEventHandler() in her testing app (And she name it as "pushHandler". Please see the below link) . But she has not yet implemented any logic in it. I'm guessing it is because we can't actually test this handlePush() at this point. 
(https://github.com/knvsl/azure-sdk-for-ios/blob/knvsl/test-app-push-notifications/examples/AzureCommunicationChatPushNotificationTest/AzureCommunicationChatPushNotificationTest/AppDelegate.swift)